lib/ProductOpener/Ingredients.pm

```diff
@@ -1290,16 +1313,14 @@ sub parse_ingredients_text ($product_ref) {
 		$ignore_strings_after_percent = $ignore_strings_after_percent{$product_lc};
 	}
 
-	my $percent_regexp
-		= '(?:<|'
-		. $min_regexp
-		. '|\s|\.|:)*(\d+(?:(?:\,|\.)\d+)?)\s*(?:\%|g)\s*(?:'
-		. $min_regexp . '|'
-		. $ignore_strings_after_percent
-		. '|\s|\)|\]|\}|\*)*';
+	# Regular expression to find percent or quantities
+	# $percent_or_quantity_regexp has 2 capturing group: one for the number, and one for the % sign or the unit
+	my $percent_or_quantity_regexp = '(?:<|' . $min_regexp . '|\s|\.|:)*' . '(\d+(?:(?:\,|\.)\d+)?)\s*'    # number
+		. '(\%|g|gr|mg|kg|ml|cl|dl|l)\s*'    # % or unit
+		. '(?:' . $min_regexp . '|' . $ignore_strings_after_percent . '|\s|\)|\]|\}|\*)*';
```

Comment:
what about using a ['x' modifier](https://perldoc.perl.org/perlre#/x-and-/xx) to make this kind of expression easier to understand ?